src/dbnode/generated/thrift/rpc.thrift

```diff
@@ -312,6 +314,41 @@ struct HealthResult {
 	2: required string status
 }
 
+enum AggregateQueryType {
+	AGGREGATE_BY_TAG_NAME,
+	AGGREGATE_BY_TAG_NAME_VALUE,
+}
+
+// AggregateQueryRequest comes from our desire to aggregate on incoming data.
+// We currently only support retrieval of facets, but could extend this based on
+// requirements in the future. Currently, the two predominant use-cases are:
+// (1) Given a filter query (optionally), return all known tag keys matching this restriction
```

Comment:
This comment is a little incorrect since `Query` is not an optional case (the no filter case would be something like `.*` which is still a query, rather than a nil)

Reply:
Good catch, made `Query` optional